File: Day 042/2810. Faulty Keyboard.cpp
class Solution {
public:
    string reverseStr(string& str) {
        int n = str.length();
        for (int i = 0; i < n / 2; i++)
            swap(str[i], str[n - i - 1]);
        return str;
    }
    
    string finalString(string s) {
        string s1;
        for (int i = 0; i < s.size(); i++) {
            if (s[i] == 'i') {
                s1 = reverseStr(s1);
            } else {
                s1.push_back(s[i]);
            }
        }
        return s1;
    }
};

File: Day 042/2810. Faulty Keyboard.java
class Solution {
    public String reverseStr(String str) {
        char[] charArray = str.toCharArray();
        int n = charArray.length;
        for (int i = 0; i < n / 2; i++) {
            char temp = charArray[i];
            charArray[i] = charArray[n - i - 1];
            charArray[n - i - 1] = temp;
        }
        return new String(charArray);
    }

    public String finalString(String s) {
        StringBuilder s1 = new StringBuilder();
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == 'i') {
                s1 = new StringBuilder(reverseStr(s1.toString()));
            } else {
                s1.append(s.charAt(i));
            }
        }
        return s1.toString();
}
}